src/java/org/apache/cassandra/db/compaction/CompactionTask.java

```diff
@@ -171,7 +177,45 @@ public boolean apply(SSTableReader sstable)
             long inputSizeBytes;
             long timeSpentWritingKeys;
 
+            List<Index> indexesReceptiveToFullyExpiredSSTables = cfs.indexManager.listIndexes()
+                                                                                 .stream()
+                                                                                 .filter(Index::includeFullyExpiredTablesInCompaction)
+                                                                                 .collect(Collectors.toList());
+
+            if (!indexesReceptiveToFullyExpiredSSTables.isEmpty() && !fullyExpiredSSTables.isEmpty())
```

Comment:
Nit: I'd check `fullyExpiredSSTables` first to avoid going through the indexes if it is empty.